fiaas_deploy_daemon/crd/status.py

```diff
@@ -45,16 +48,49 @@ def now():
     return now.isoformat()
 
 
-def _handle_signal(sender, status, subject):
+def _handle_signal_without_status(sender, status, subject):
+    if status == STATUS_STARTED:
+        status = "RUNNING"
+    else:
+        status = status.upper()
+
+    _save_status(status, subject)
+    _cleanup(subject.app_name, subject.namespace)
+
+
+def _handle_signal_with_status(sender, status, subject):
     if status == STATUS_STARTED:
         status = "RUNNING"
     else:
         status = status.upper()
 
+    _save_status_inline(status, subject)
     _save_status(status, subject)
     _cleanup(subject.app_name, subject.namespace)
 
 
+@retry_on_upsert_conflict
+def _save_status_inline(result, subject):
+    (uid, app_name, namespace, deployment_id, repository, labels, annotations) = subject
+
+    app = FiaasApplication.get(app_name, namespace)
+    generation = int(app.metadata.generation)
+    try:
+        application_deployment_id = app.metadata.labels["fiaas/deployment_id"]
+    except (AttributeError, KeyError, TypeError):
+        raise ValueError("The Application {} is missing the 'fiaas/deployment_id' label".format(app_name))
+    # We only want to get error logs here.
+    if deployment_id == application_deployment_id:
+        logs = _get_error_logs(app_name, namespace, deployment_id, result)
+
+        LOG.info("Saving inline result %s for %s/%s generation %s", result, namespace, app_name, generation)
```

Comment:
nitpick: I often find the deployment_id part of the "Saving result" log message from `_save_status` useful when troubleshooting, so I think it could be helpful to include that here as well
```suggestion
        LOG.info("Saving inline result %s for %s/%s deployment_id=%s generation %s", result, namespace, app_name, deployment_id, generation)
```